storage/src/rocksdb_client.cpp

```diff
@@ -429,6 +432,11 @@ KeyValuePair ClientIterator::next() {
     LOG_DEBUG(logger, "Reading count = " << g_rocksdb_called_read);
   }
 
+  if (!m_iter->Valid()) {
+    LOG_ERROR(logger, "Iterator is not valid");
+    m_status = Status::GeneralError("Iterator is not valid");
```

Comment:
Why do we set the m_status here, but not in `previous` ?

Reply:
I noticed it too. I am not really sure, I suspect it was just missed. However, m_status is not being used anywhere. Therefore, I will also set it here.